sanityv3/src/lib/structure/index.js

```diff
@@ -55,45 +55,28 @@ const SUB_EDITOR_ITEMS = (S) =>
 
 const LOCAL_NEWS_EDITOR_ITEMS = (S) => [LocalNews(S)].filter((e) => e)
 
+/**
+ * Datasets are not taken into consideration in order to simplify logic.
+ * This is used only to hide/show items in the UI.
+ *
+ * Drawback scenario:
+ *  An user with "Editor" access to the "brazil" dataset, and "Sub Editor" to the "argentina" dataset
+ *  will get ADMIN_ITEMS menu structure for both studios. This is acceptable because it is just
+ *  the UI menu items, the content is properly hidden and handled at sanity.io -> Access tab.
+ */
+
 const getItems = (S, context) => {
-  /**
-   * Datasets are not taken into consideration in order to simplify logic.
-   * This is used only to hide/show items in the UI.
-   *
-   * Drawback scenario:
-   *  An user with "Editor" access to the "brazil" dataset, and "Sub Editor" to the "argentina" dataset
-   *  will get ADMIN_ITEMS menu structure for both studios. This is acceptable because it is just
-   *  the UI menu items, the content is properly hidden and handled at sanity.io -> Access tab.
-   */
   const userRoles = getCurrentUserRoles(context)
   const isAdmin = userRoles.some(
     (role) => role.startsWith('editor') || role === 'administrator' || role === 'developer',
   )
   const isSubEditor = userRoles.some((role) => role.startsWith('sub-editor'))
   const isLocalNewsEditor = userRoles.some((role) => role.startsWith('local-news-editor'))
-  return [
-    News(S),
-    LocalNews(S, context),
-    TopicContent(S),
-    LandingPage(S),
-    Event(S),
-    Magazine(S),
-    Misc(S),
-    S.divider(),
-    Homepage(S),
-    Routes(S, context),
-    S.divider(),
-    Menu(S),
-    Footer(S),
-    S.divider(),
-    AssetLibrary(S, context),
-    S.divider(),
-    Settings(S),
-  ].filter((e) => e)
+
   if (isAdmin) {
-    return ADMIN_ITEMS(S)
+    return ADMIN_ITEMS(S, context)
   } else if (isSubEditor) {
-    return SUB_EDITOR_ITEMS(S)
+    return SUB_EDITOR_ITEMS(S, context)
   } else if (isLocalNewsEditor) {
     return LOCAL_NEWS_EDITOR_ITEMS(S)
```

Comment:
is that necessary when LOCAL_NEWS_EDITOR_ITEMS does not use context? 

`const LOCAL_NEWS_EDITOR_ITEMS = (S) => [LocalNews(S)].filter((e) => e)
`

Reply:
yes, it is wrong because it should use context, since LocalNews use context:

`const LocalNews = (S, context) => Flags.HAS_LOCAL_NEWS && LocalNewsStructure(S, context)`